examples/project/Directory.Build.targets

```diff
@@ -0,0 +1,20 @@
+<Project>
+
+  <!--
+    To enable out-of-this-repository tests, we need to rely on internally-built contracts only during our tests.
+    If anyone else wants to run these, they should be able to do so with a public NuGet package.
+
+    Since we don't really want to change generator API just for tests (which would be quite painful even if we wanted
+    to), we rely on environment variables that mark that we are running in the inner test suite and we have the
+    `LeanCode.Contracts` package already built.
+  -->
+
+  <ItemGroup Condition=" '$(UseTestBuildOfContracts)' == 'true' ">
+    <PackageReference Update="LeanCode.Contracts" Version="9.9.9.9-internal" />
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(UseTestBuildOfContracts)' != 'true' ">
+    <PackageReference Update="LeanCode.Contracts" Version="2.0.0-preview.3" />
+  </ItemGroup>
```

Comment:
Maybe wrapping it with
```xml
<Choose>
    <When Condition=" ... ">
        ...
    </When>
    <Otherwise>
        ...
    </Otherwise>
</Choose>
```
would be slightly less ugly?

Reply:
Another approach could be storing a version number in a custom property and providing an override.

Reply:
TIL that `Choose` exists.

Nevertheless, I don't really think this changes much for a single group, neither the conditional (it will be more code) nor the version (it will be awkward and even more "distributed" - the version would be in 2 very distinct places then).

Reply:
TIL x2, I was browsing docs, wondering if there is some if-else construct. :P
Anyway, I'd argue that `Choose` would be better in a way that it says explicitly that the `ItemGroups` are related, without (a bit awkward) condition repetition. This is a nitpick though, so do as you prefer ofc.